_posts/2019-10-01-intro-to-hmc/hmc-intro.Rmd

```diff
@@ -0,0 +1,283 @@
+If after reading, you still feel confused, it isn't your fault and _maybe_, not even the fault of this text. Based on an inofficial, minimal-sample-size survey, this topic is a _lot_ more confusing than deep learning :-). As to the content, we'll try to keep it as digestable as possible, while at the same time striving to mention all of the famous incantations: Hamiltonian Monte Carlo, _leapfrog_ steps, NUTS.
+
+
+## So what is it for?
+
+Sampling, or _Monte Carlo_, techniques in general are used when we want to produce samples from, or statistically describe, a distribution we don't have a closed-form formulation of. Sometimes, we might really be interested in the samples; sometimes we just want them so we can compute, for example, the mean and variance of the distribution.
+
+What distribution? In the type of applications we're talking about, we have a _model_, a joint distribution, supposed to describe some reality. Starting from the most basic scenario, it might look like this:
+
+$$
+x \sim \mathcal{Poisson}(\lambda)
+$$
+
+This "joint distribution" only has a single member, a Poisson distribution, that is supposed to model, say, the number of comments in a code review. We also have data on actual code reviews, like this, say ^[data are purely made up]:
+
+```{r, echo = FALSE}
+ggplot(data.frame(x = x), aes(x = x)) + geom_bar() + ggtitle("Number of comments in (fictional) code reviews") + labs(x = "number of comments", y = "number of code reviews") + scale_x_discrete(drop=FALSE)
+```
+
+```{r, eval=TRUE, echo=FALSE, layout="l-body", fig.cap = ""}
+knitr::include_graphics("images/comments.png")
+```
+
+We now want to determine the _parameter_, $\lambda$, of the Poisson that make these data most _likely_. So far, we're not even beign Bayesian yet: There is no prior on this parameter. But of course, we want to be Bayesian, so we add one -- imagine fixed priors ^[elided] on _its_ parameters:
+
+$$
+x \sim \mathcal{Poisson}(\lambda)\\
+\lambda \sim \gamma(\alpha, \beta)\\
+\alpha \sim [...]\\  
+\beta \sim [...]
+$$
+
+This now really being a joint distribution, we have three parameters to determine: $\lambda$, $\alpha$ and $\beta$.
+And what we're interested in is the _posterior distribution_ of the parameters given the data.
+
+Now many expositions will tell you how, depending on the distributions involved, we usually cannot calculate the posterior distributions in closed form. Instead of reiterating these explanations, here we'll just build on them and state that we have to use sampling techniques to determine those parameters. ^[In some cases, variational inference is an alternative. See [@Blei_2017] for a nice introduction.] What we'd like to point out instead is the following: In the upcoming discussions of sampling, HMC & co., it is really easy to forget _what is it that we are sampling_. Try to always keep in mind that what we're sampling isn't the data, it's parameters: the parameters of the posterior distributions we're interested in.
```

Comment:
> Now many expositions will tell you how, depending on the distributions involved, we usually cannot calculate the posterior distributions in closed form.

Could this be reduced to something like: Usually, we cannot calculate the posterior distributions in closed form. ?

Then you have the ~ "see a bunch of other things if you want to understand this" afterward.